src/gridcoin/beacon.cpp

```diff
@@ -989,6 +997,35 @@ int BeaconRegistry::BeaconDB::Initialize(PendingBeaconMap& m_pending, BeaconMap&
 {
     bool status = true;
     int height = 0;
+    uint32_t version = 0;
+
+    // First load the beacon db version from leveldb and check it against the constant in the class.
+    {
+        CTxDB txdb("r");
+
+        std::pair<std::string, std::string> key = std::make_pair("beacon_db", "version");
+
+        bool status = txdb.ReadGenericSerializable(key, version);
+
+        if (!status) version = 0;
+    }
+
+    if (version != m_version)
```

Comment:
I don't see why we need the `BeaconDB::m_version` member field. It looks like we want to check... 

```c++
if (version != CURRENT_VERSION) { ... }
```

...and set the version in `clear_leveldb()` with...

```c++
status &= txdb.WriteGenericSerializable(key, CURRENT_VERSION);
```

Is there a reason for the mutable state? Maybe I'm missing something...

Reply:
I originally had it that way and was getting linker errors.

Reply:
I will try and change it back.

Reply:
Your suggestion to change static constexpr to static const worked. !@$$% linker...

Reply:
Fixed.